graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/hooks/SchemaGeneratorHooks.kt

```diff
@@ -90,22 +100,58 @@ interface SchemaGeneratorHooks {
     /**
      * Called after wrapping the type based on nullity but before adding the generated type to the schema
      */
-    fun didGenerateGraphQLType(type: KType, generatedType: GraphQLType) = generatedType
+    @Suppress("Detekt.ThrowsCount")
+    fun didGenerateGraphQLType(type: KType, generatedType: GraphQLType): GraphQLType {
```

Comment:
we are validating in `didGenerateGraphQLType` instead of `willAddGraphQLTypeToSchema` hook to allow custom modifications in the latter (including adding new fields).